var/spack/repos/builtin/packages/xz/package.py

```diff
@@ -61,7 +68,67 @@ def determine_version(cls, exe):
         match = re.search(r"xz \(XZ Utils\) (\S+)", output)
         return match.group(1) if match else None
 
+
+class AutotoolsBuilder(AutotoolsBuilder):
+    def configure_args(self):
+        return self.enable_or_disable("libs")
+
     @run_after("install")
     def darwin_fix(self):
         if self.spec.satisfies("platform=darwin"):
             fix_darwin_install_name(self.prefix.lib)
+
+
+class MSBuildBuilder(MSBuildBuilder):
+    @property
+    def build_directory(self):
+        def get_file_string_number(f):
+            s = re.findall(r"\d+$", f)
+            return (int(s[0]) if s else -1, f)
+
+        win_dir = os.path.join(super().build_directory, "windows")
+        compiler_dirs = []
+        with working_dir(win_dir):
+            for obj in os.scandir():
+                if obj.is_dir():
+                    compiler_dirs.append(obj.name)
+        newest_compiler = max(compiler_dirs, key=get_file_string_number)
+        return os.path.join(win_dir, newest_compiler)
+
+    @property
+    def toolchain_version(self):
+        return "v" + self.pkg.compiler.platform_toolset_ver
```

Comment:
I don't see where this is used: is this method required for this builder to succeed?

Reply:
It looks like by default

```
    @property
    def toolchain_version(self):
        return self.compiler.msvc_version

    @property
    def msvc_version(self):
        return Version(re.search(Msvc.version_regex, self.cc).group(1))
```

Which appears to be unchanging. Or does the value of `.cc` change? I'm not clear on:

* How this is a consistency error
* Why this isn't automatically-detectable and fixable as part of the MSBuild class itself

Reply:
Sorry, for whatever reason I _badly_ misread your comment and thought you were questioning why the property was needed in the first place. 

This has nothing to do with consistency and everything to do with how we organized these PRs. Looking back in my notes this should be removed and the actual MSbuild builder patched to perform this operation now that #35098 has landed.

Reply:
> Sorry, for whatever reason I badly misread your comment and thought you were questioning why the property was needed in the first place.

To be clear I think your original reading of https://github.com/spack/spack/pull/35095#discussion_r1096461931 was correct: I indeed missed that `toolchain_version` is used in the `MSBuild` class. That being said, my response in https://github.com/spack/spack/pull/35095#discussion_r1097945726 raises an additional concern about clarity; it sounds like you agree with that part but if that leaves anything unclear let me know.